### PR TITLE
Bug fixes & tests for ExperimentRowCard

### LINKED ISSUE #1541: Backfill tests for ExperimentRowCard component
This is a leftover from #1399. The ExperimentRowCard component still needs tests.

### PR DESCRIPTION
**Pull request by lmorchard**
- Restore "last seen" timestamp in localStorage set when viewing an
  experiment page, lost in translation from Ampersand.
- Bugfix to properly handle "last seen" timestamp in relation to "just
  launched" and "just updated" banners
- Rework ExperimentCardList-test to squelch warnings about missing
  required props in testing
- Add l10n strings for hardcoded ending soon status messages
- Tests

Fixes #1541

### CODE REVIEW

File: frontend/src/app/components/ExperimentPage.js [hunk 1]
```diff
@@ -147,6 +147,10 @@ export default class ExperimentPage extends React.Component {
             introduction, measurements, privacy_notice_url, changelog_url,
             thumbnail, survey_url, contributors, details, min_release } = experiment;
 
+    // Set the timestamp for when this experiment was last seen (for
+    // ExperimentRowCard updated/launched banner logic)
+    setExperimentLastSeen(experiment);
```
Comment (lmorchard) [inline frontend/src/app/components/ExperimentPage.js]: We lost this from ye olde Ampersand site, [over here](https://github.com/mozilla/testpilot/blob/2016-09-12/frontend/src/app/views/experiment-page.js#L272). Just realized it wasn't working as I wrote the test.

File: frontend/src/app/components/ExperimentRowCard.js [hunk 2]
```diff
@@ -81,14 +81,14 @@ export default class ExperimentRowCard extends React.Component {
     if ((now - modified) > MAX_JUST_UPDATED_PERIOD) { return false; }
 
     // If modified since the last time seen, *do* consider it updated.
-    if (modified > experiment.lastSeen) { return true; }
+    if (modified > getExperimentLastSeen(experiment)) { return true; }
```
Comment (lmorchard) [inline frontend/src/app/components/ExperimentRowCard.js:96]: We lost this from ye olde Ampersand site, [over here](https://github.com/mozilla/testpilot/blob/2016-09-12/frontend/src/app/models/experiment.js#L44). Just realized it wasn't working as I wrote the test.

File: frontend/test/app/components/ExperimentCardList-test.js [hunk 3]
```diff
@@ -1,34 +1,41 @@
...
-    expect(wrapper.contains(<Loading />)).to.equal(true);
+  let props, subject;
+  beforeEach(() => {
+    props = {
+      experiments: [
+        { slug: 'foo' },
+        { slug: 'bat' }
+      ],
+      isExperimentEnabled: sinon.spy(() => false),
+      // required by ExperimentRowCard {...this.props}
```
Comment (lmorchard) [inline frontend/test/app/components/ExperimentCardList-test.js:18]: Not super happy with these shenanigans, but not sure how else to manage props expected by child components via `{...props}`

**Comment by dannycoates** [comment]
Rebase? I'd like to see `setExperimentLastSeen` get a propType on `ExperimentPage`

**Comment by lmorchard** [comment]
Rebased & added `setExperimentLastSeen` as a propType on `ExperimentPage`

**Review by dannycoates — APPROVED** [review]
LGTM

**Comment by dannycoates** [comment]
> LGTM

of course, eslint doesn't think so 😉

**Comment by lmorchard** [comment]
Rebased again and excised my extra space

**Pull request merged**